app/services/reporter.rb

```diff
@@ -6,61 +6,80 @@
 # run queries and produce reports from the results, for consumption
 # by preservation catalog maintainers
 class Reporter
-  attr_reader :storage_root, :druids
+  attr_reader :storage_root
 
   # @params [Hash] params used to initialize the Reporter service
   # @return [Reporter] the reporter
   def initialize(params)
     @storage_root = MoabStorageRoot.find_by!(name: params[:storage_root_name])
-    @druids = moab_storage_root_druid_list
+    @msr_names = {}
   end
 
-  # @return [Array] an array of druids on the storage root
-  def moab_storage_root_druid_list
-    druid_array = []
+  def moab_storage_root_list_pres_obj_comp_moabs_relation
     PreservedObject
       .joins(:complete_moabs)
       .where(complete_moabs: { moab_storage_root: storage_root })
-      .select(:druid)
       .order(:druid)
+  end
+
+  # @return [Array] an array of druids on the storage root
+  def druid_csv_list
+    druid_array = [['druid']]
+    moab_storage_root_list_pres_obj_comp_moabs_relation
+      .select(:druid)
       .each_row do |po_hash|
-        druid_array << po_hash['druid']
+        druid_array << [po_hash['druid']]
       end
     druid_array
   end
 
-  # @param [Array] druids - list of druids to output details for
   # @param [Boolean] errors_only (default: false) - optionally only output lines with audit errors
   # @return [Array] an array of hashes with details for each druid provided
```

Comment:
Is this now returning an array of arrays? How about `# @return [Array<Array>] an array of arrays, with each inner array representing a CSV row (including a header row)` ?

Reply:
ah, yes!  good catch.